frontend/src/lib/utils.tsx

```diff
@@ -79,8 +79,19 @@ export function toParams(obj: Record<string, any>): string {
         val = typeof val === 'object' ? JSON.stringify(val) : val
         return encodeURIComponent(val)
     }
+
     return Object.entries(obj)
         .filter((item) => item[1] != undefined && item[1] != null)
+        .filter((item) => item[1].length > 0)
+        .reduce((acc, [key, val]) => {
+            if (explodeArrays && Array.isArray(val)) {
```

Comment:
I'm not normally a fan of boolean parameters to change behaviour like this but I needed different behaviour than this method has been exposing and didn't want to randomly break other areas

Reply:
Whenever I see `reduce` code just loses all readability, and I need to go into code parsing mode to make sure I understand it fully. Perhaps a comment here could help? It's not inherently obvious what does an array look like after it exploded.

I think this `if` would look cleaner in the next, `map` block with a nested `map` to be honest, skipping this entire function.